### ISSUE TITLE
Appliances which require a domain to be set, should auto redirect IP access to the domain

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#958]
This issue is to consolidate issues #869 & #871 as well as highlight the other appliances which need to be checked. Here are all the appliances which require a domain to be set (as of v14.2):

- [ ] canvas
- [x] django - #869 (not tagged in this issue)
- [x] drupal8
- [x] elgg (should be ok - see below)
- [x] ghost
- [ ] gitlab
- [x] gnusocial
- [x] icescrum
- [x] magento
- [x] mediawiki
- [x] oscommerce - #1136 (not tagged in this issue)
- [x] phplist
- [ ] piwik
- [x] prestashop
- [x] punbb
- [x] simplemachines
- [x] zencart - #871 (not tagged in this issue)

Without digging into the code, I can't be 100% clear that they will necessarily all need to be updated, but they will need to be tested. If the main site doesn't work as it should when contacting via IP (e.g. http://xxx.xxx.xxx.xxx) address, we should be redirecting to the domain.

FWIW, the below appliances have Apache redirect configured, so may provide some useful hints?! They all redirect to HTTPS, but that shouldn't be required (but is probably desirable):

- owncloud: redirects to https - [owncloud/overlay/etc/apache2/sites-available/owncloud.conf](https://github.com/turnkeylinux-apps/owncloud/tree/master/overlay/etc/apache2/sites-available/owncloud.conf#L8-L9)
- mahara: redirects to https - [mahara/overlay/etc/apache2/sites-available/mahara.conf](https://github.com/turnkeylinux-apps/mahara/blob/master/overlay/etc/apache2/sites-available/mahara.conf#L6-L8)
- nextcloud: redirects to https (same as ownCloud) - [nextcloud/overlay/etc/apache2/sites-available/nextcloud.conf](https://github.com/turnkeylinux-apps/nextcloud/blob/master/overlay/etc/apache2/sites-available/nextcloud.conf#L8-L9)
- elgg: redirects to https - [elgg/overlay/etc/apache2/sites-available/elgg.conf](https://github.com/turnkeylinux-apps/elgg/blob/master/overlay/etc/apache2/sites-available/elgg.conf#L5-L7)

**Comment by l-arnold** [comment]
Easy to have Magento and Joomla redirect to Https.  I believe the Odoo Appliance was redirecting to Https out of the box.

________________________________

**Comment by JedMeister** [comment]
@spaghettimaster - I was about to remove you from this issue, as the 3 remaining apps are canvas, gitlab (@qq7) and pwik (@OnGle). However, perhaps they would like your assistance with them?

I'm almost certain that @OnGle would be happy for you to take care of piwik (soon to be renamed matomo - see #1003) but we're still blocked by the fact that there isn't a matomo Debian package.

**Comment by JedMeister** [comment]
Added oscommerce.

**Comment by JedMeister** [comment]
Added mediawiki to the list and ticked it off straight away - resolved via https://github.com/turnkeylinux-apps/mediawiki/pull/13

**Comment by JedMeister** [comment]
This should be pretty much done now so closing. If any are actually outstanding, or there are new ones, best to open a specific new issue for them.
